File: unit_testing_fundamentals/src/test/java/com/techyourchance/unittestingfundamentals/example2/StringReverserTest.java
package com.techyourchance.unittestingfundamentals.example2;

import static org.hamcrest.CoreMatchers.is;
import static org.junit.Assert.assertThat;

import org.junit.Before;
import org.junit.Test;

public class StringReverserTest {

    StringReverser SUT;

    @Before
    public void setup() throws Exception {
        SUT = new StringReverser();
    }

    @Test
    public void reverse_emptyString_emptyStringReturned() throws Exception {
        String result = SUT.reverse("");
        assertThat(result, is(""));
    }

    @Test
    public void reverse_singleCharacter_sameStringReturned() throws Exception {
        String result = SUT.reverse("a");
        assertThat(result, is("a"));
    }

    @Test
    public void reverse_longString_reversedStringReturned() throws Exception {
        String result = SUT.reverse("Vasiliy Zukanov");
        assertThat(result, is("vonakuZ yilisaV"));
    }
}

File: unit_testing_fundamentals/src/test/java/com/techyourchance/unittestingfundamentals/example2/StringReverserTest.kt
package com.techyourchance.unittestingfundamentals.example2

import org.junit.Assert.assertEquals
import org.junit.Test

class StringReverserTest {

    private val reverser = StringReverser()

    @Test
    @Throws(Exception::class)
    fun reverse_emptyString_emptyStringReturned() {
        assertEquals("", reverser.reverse(""))
    }

    @Test
    @Throws(Exception::class)
    fun reverse_singleCharacter_sameStringReturned() {
        assertEquals("a", reverser.reverse("a"))
        assertEquals("1", reverser.reverse("1"))
        assertEquals("$", reverser.reverse("$"))
    }

    @Test
    @Throws(Exception::class)
    fun reverse_longString_reversedStringReturned() {
        assertEquals("vonakuZ yilisaV", reverser.reverse("Vasiliy Zukanov"))
        assertEquals("ahihT gnuA", reverser.reverse("Aung Thiha"))
        assertEquals("eoS irihT nihK", reverser.reverse("Khin Thiri Soe"))
    }
}